pyrdp/mitm/X224MITM.py

```diff
@@ -99,6 +99,12 @@ def onConnectionConfirm(self, _: X224ConnectionConfirmPDU):
         protocols = NegotiationProtocols.SSL if self.originalRequest.tlsSupported else NegotiationProtocols.NONE
 
         parser = NegotiationResponseParser()
+        response = parser.parse(pdu.payload)
+        if response.type == NegotiationType.TYPE_RDP_NEG_FAILURE:
+            self.log.info("The server failed the negotiation. Error: %(error)s", {"error": NegotiationFailureCode.getMessage(response.failureCode)})
+            return
+
+        # TODO: cleanup
```

Comment:
Maybe you meant exit the program differently? Simply returning in this scenario seems odd 🤔

Reply:
Yeah.. as discussed, we forgot and the change has is should be beneficial so let's go ahead.